File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] nums) {
        int maxi = Integer.MIN_VALUE;
        int start = 0, end = nums.length-1;

        while(start<end){
            if(nums[start] < nums[end]){
                maxi = Math.max(maxi, (end-start) * nums[start]);
                start++;
            }
            else {
                maxi = Math.max(maxi, (end-start) * nums[end]);
                end--;
            }
        }

        return maxi;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        start = 0
        end = len(height) - 1

        while start < end:
            if height[start] < height[end]:
                ans = max(ans, (end-start) * height[start])
                start += 1
            else:
                ans = max(ans, (end-start) * height[end])
                end -= 1

        return ans
